src/services/device.js

```diff
@@ -0,0 +1,110 @@
+import { DATE_FORMAT_KEY, airnoteProductUID } from '../constants';
+import { format } from 'date-fns';
+import queryString from 'query-string';
+
+function getHistory(readings) {
```

Comment:
You could probably have elasticsearch do more of this work for you with things like https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-datehistogram-aggregation.html

But probably not a huge benefit at this scale. Maybe slighly faster/smaller page loads.

Reply:
I wish I would’ve stumbled upon this when I was first tinkering with this query 😅

This definitely seems cleaner. I’m going to hold off messing with this for now (just because I don’t want to hack with things this close to releasing v1), but I’ve got that doc page bookmarked now.

This app’s main query is only going to get more complex as we add stuff to the dashboard, so moving some of the work to elasticsearch seems smart (and cleaner) 🙂

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html is the top level docs, there's a _ton_ of options under the 3 agg categories (bucket, metric, pipeline).